dapp/views/smartlocker/subviews/create_or_link/create_or_link.jsx

```diff
@@ -4,16 +4,18 @@ import { bindActionCreators }      from 'redux'
 import { Button, Input, Blipping } from '@poplocker/react-ui'
 import { RegistrarContract }       from 'lib/contracts'
 import { rpc }                     from 'lib/rpc_calls'
-import { flags }                     from 'lib/helpers'
+import { flags }                   from 'lib/helpers'
 
 import './create_or_link.css'
 
 class CreateOrLinkSubview extends React.Component {
   constructor (props) {
     super(props);
-    // TODO: address of the registrar
-    // should be provided by the extension
-    this.registrar = new RegistrarContract(config.contracts.registrar);
+
+    const { abi } = config.contracts.registrar;
+    const { address } = props.locker.registrar;
+
+    this.registrar = new RegistrarContract({ abi, address });
```

Comment:
I think at this point calling `new RegistrarContract(abi, address)` would be nicer. That requires updating `contracts.js` with something like:

```
constructor(abi, address)
```
and 
```
super(abi, address)
```

Reply:
Done